<?xml version="1.0" encoding="utf-8"?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from /home/joemasterjohn/hsr/src/hsr_description/robots/hsrb4s.urdf.xacro | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<!--
Copyright (c) 2017, Toyota Motor Corporation
All rights reserved.

Redistribution and use in source and binary forms, with or without
modification, are permitted provided that the following conditions are met:

 * Redistributions of source code must retain the above copyright notice,
   this list of conditions and the following disclaimer.
 * Redistributions in binary form must reproduce the above copyright
   notice, this list of conditions and the following disclaimer in the
   documentation and/or other materials provided with the distribution.
 * Neither the name of Toyota Motor Corporation nor the names of its
   contributors may be used to endorse or promote products derived from
   this software without specific prior written permission.

THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
POSSIBILITY OF SUCH DAMAGE.
-->
<robot name="hsrb" xmlns:controller="http://playerstage.sourceforge.net/gazebo/xmlschema/#controller" xmlns:interface="http://playerstage.sourceforge.net/gazebo/xmlschema/#interface" xmlns:sensor="http://playerstage.sourceforge.net/gazebo/xmlschema/#sensor">
  <material name="body_main">
    <color rgba="1.0 0.0 0.0 1.0"/>
  </material>
  <material name="body_sub">
    <color rgba="0.3333 0.3333 0.3333 1.0"/>
  </material>
  <material name="wheel">
    <color rgba="0.2 0.2 0.2 1.0"/>
  </material>
  <material name="black">
    <color rgba="0.1 0.1 0.1 1.0"/>
  </material>
  <material name="tablet">
    <color rgba="0.1 0.1 0.2 1.0"/>
  </material>
  <!--
    ASUS Xtion PRO LIVE
    Specs from:
    http://www.asus.com/Multimedia/Xtion_PRO_LIVE/#specifications
    -->
  <link name="base_footprint"/>
  <joint name="base_footprint_joint" type="fixed">
    <parent link="base_footprint"/>
    <child link="base_link"/>
  </joint>
  <link name="base_link">
    <inertial>
      <mass value="50.00"/>
      <!-- use dummy weight to stabilize gazebo move base (real value: 11.017971)-->
      <origin xyz="-0.025978 -0.005498 0.17633"/>
      <inertia ixx="0.3742" ixy="0.000434172172" ixz="0.03088" iyy="0.3436" iyz="0.01775" izz="0.1509"/>
    </inertial>
    <visual>
      <geometry>
        <mesh filename="./meshes/base_v2/base_light.obj"/>
      </geometry>
    </visual>
    <visual>
      <geometry>
        <mesh filename="./meshes/base_v2/body_light.obj"/>
      </geometry>
    </visual>
    <visual>
      <geometry>
        <mesh filename="./meshes/base_v2/torso_base.obj"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="./meshes/base_v2/base.collision.obj"/>
      </geometry>
    </collision>
  </link>
  <joint name="arm_lift_joint" type="prismatic">
    <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.340"/>
    <axis xyz="0.0 0.0 1.0"/>
    <limit effort="100.0" lower="0.0" upper="0.69" velocity="0.2"/>
    <parent link="base_link"/>
    <child link="arm_lift_link"/>
    <dynamics damping="1.0"/>
  </joint>
  <link name="arm_lift_link">
    <inertial>
      <mass value="1.275819"/>
      <origin xyz="0.051549 0.042438 -0.064145"/>
      <inertia ixx="0.004633" ixy="0.000475272668" ixz="-0.002403" iyy="0.008065" iyz="-0.000100268629" izz="0.007672"/>
    </inertial>
    <visual>
      <geometry>
        <mesh filename="./meshes/arm_v0/shoulder.obj"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="./meshes/arm_v0/shoulder.collision.obj"/>
      </geometry>
    </collision>
  </link>
  <joint name="arm_flex_joint" type="revolute">
    <origin rpy="0.0 0.0 0.0" xyz="0.141 0.078 0.0"/>
    <axis xyz="0.0 -1.0 0.0"/>
    <limit effort="100.0" lower="-2.62" upper="0.0" velocity="1.2"/>
    <parent link="arm_lift_link"/>
    <child link="arm_flex_link"/>
    <dynamics damping="1.0"/>
  </joint>
  <link name="arm_flex_link">
    <inertial>
      <mass value="1.339295"/>
      <origin xyz="-0.000459 0.006583 0.106069"/>
      <inertia ixx="0.00889053" ixy="0.0" ixz="0.0" iyy="0.00825549" iyz="0.0" izz="0.00190511"/>
    </inertial>
    <visual>
      <geometry>
        <mesh filename="./meshes/arm_v0/arm_flex_light.obj"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="./meshes/arm_v0/arm_flex.collision.obj"/>
      </geometry>
    </collision>
  </link>
  <joint name="arm_roll_joint" type="revolute">
    <origin rpy="0.0 0.0 0.0" xyz="0.005 0.0 0.345"/>
    <axis xyz="0.0 0.0 1.0"/>
    <limit effort="100.0" lower="-2.09" upper="3.84" velocity="2.0"/>
    <parent link="arm_flex_link"/>
    <child link="arm_roll_link"/>
    <dynamics damping="0.1"/>
  </joint>
  <link name="arm_roll_link">
    <inertial>
      <mass value="0.247936"/>
      <origin xyz="-0.006927 0.002039 -0.064837"/>
      <inertia ixx="0.004267" ixy="-0.000018322199" ixz="0.000017984455" iyy="0.000780923828" iyz="-0.00001392334" izz="0.00412"/>
    </inertial>
    <visual>
      <geometry>
        <mesh filename="./meshes/arm_v0/arm_roll_light.obj"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="./meshes/arm_v0/arm_roll.collision.obj"/>
      </geometry>
    </collision>
  </link>
  <joint name="wrist_flex_joint" type="revolute">
    <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
    <axis xyz="0.0 -1.0 0.0"/>
    <limit effort="100.0" lower="-1.92" upper="1.22" velocity="1.5"/>
    <parent link="arm_roll_link"/>
    <child link="wrist_flex_link"/>
    <dynamics damping="1.0"/>
  </joint>
  <link name="wrist_flex_link">
    <inertial>
      <mass value="0.681225"/>
      <origin xyz="0.000131 -0.003929 0.00217"/>
      <inertia ixx="0.000074353558" ixy="0.000000494515" ixz="0.000000022937" iyy="0.000038432887" iyz="-0.00000255576" izz="0.000064680918"/>
    </inertial>
    <visual>
      <geometry>
        <mesh filename="./meshes/wrist_v0/wrist_flex.obj"/>
      </geometry>
      <material name="body_sub"/>
    </visual>
    <visual>
      <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.04"/>
      <geometry>
        <cylinder length="0.02" radius="0.017"/>
      </geometry>
      <material name="body_sub"/>
    </visual>
    <collision>
      <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.03"/>
      <geometry>
        <cylinder length="0.04" radius="0.017"/>
      </geometry>
    </collision>
  </link>
  <joint name="wrist_roll_joint" type="revolute">
    <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
    <axis xyz="0.0 0.0 1.0"/>
    <limit effort="100.0" lower="-1.92" upper="3.67" velocity="1.5"/>
    <parent link="wrist_flex_link"/>
    <child link="wrist_roll_link"/>
    <dynamics damping="1.0"/>
  </joint>
  <link name="wrist_roll_link">
    <inertial>
      <mass value="0.761692"/>
      <origin xyz="-0.001416 -0.001547 0.081689"/>
      <inertia ixx="0.000805633819" ixy="-0.000014431758" ixz="0.000028204659" iyy="0.000768077165" iyz="0.000022460226" izz="0.000453406611"/>
    </inertial>
    <visual>
      <geometry>
        <mesh filename="./meshes/wrist_v0/wrist_roll.obj"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="./meshes/wrist_v0/wrist_roll.collision.obj"/>
      </geometry>
    </collision>
  </link>
  <joint name="hand_palm_joint" type="fixed">
    <origin rpy="0.0 0.0 3.14159265359" xyz="0.012 0.0 0.1405"/>
    <parent link="wrist_roll_link"/>
    <child link="hand_palm_link"/>
  </joint>
  <link name="hand_palm_link">
    <inertial>
      <mass value="0.15"/>
      <inertia ixx="0.015" ixy="0.0" ixz="0.0" iyy="0.015" iyz="0.0" izz="0.015"/>
    </inertial>
    <visual>
      <geometry>
        <mesh filename="./meshes/hand_v0/palm.obj"/>
      </geometry>
      <material name="body_sub"/>
    </visual>
    <collision>
      <geometry>
        <mesh filename="./meshes/hand_v0/palm.obj"/>
      </geometry>
    </collision>
  </link>
</robot>

